### ISSUE TITLE
Named parsers fail to parse named headers without name

### ISSUE THREAD

**Issue by ssankko** [ByteHeathen/libsip#5]
So, as titled.
Notation like `From: sip:unknown@127.0.0.1;tag=47e9e947377149ba882adad161eacae0` is allowed in rfc, but not implemented in libsip. Parser skips and parses them like `Other` header.
Maybe changing `impl_named_parser` to something like this will fix it.
```
macro_rules! impl_named_parser {
    ($name:tt, $tag:tt, $variant:ident) => {
        named!(pub $name<Header>, do_parse!(
            tag_no_case!($tag) >>
            opt!(take_while!(is_space)) >>
            char!(':') >>
            opt!(take_while!(is_space)) >>
            res: alt!(
                do_parse!(
                    out: parse_named_field_value >>
                    params: parse_named_field_params >>
                    tag!("\r\n") >> 
                    (Header::$variant(NamedHeader { display_name: out.0, uri: out.1, params: params }))
                ) | 
                do_parse!(
                    out: parse_uri >> 
                    params: parse_named_field_params >>
                    tag!("\r\n") >> 
                    (Header::$variant(NamedHeader { display_name: None, uri: out, params: params }))
                )
            ) >> 
            (res)
        ));
    }
}
```

**Comment by bytebuddha** [comment]
I'm currently looking into this. This should be handled in `parse_named_field_value` i think actually

**Comment by bytebuddha** [comment]
Fixed in v2.4
